vumi_http_retry/retries.py

```diff
@@ -20,3 +26,26 @@ def add_request(redis, prefix, req):
     }
 
     return redis.zadd(requests_key(prefix), next_score(req), json.dumps(req))
+
+
+@inlineCallbacks
+def pop_requests(redis, prefix, from_time, to_time):
+    k = requests_key(prefix)
+    reqs = yield redis.zrangebyscore(k, from_time, to_time)
```

Comment:
`zrangebyscore` has a `LIMIT <offset> <count>` argument that it would be awesome to use here to ensure that we don't try pull thousands of items from Redis in one go (i.e. we cap the amount of work we might possibly do). Not sure if txredis supports it yet?